gui/network/forms.py

```diff
@@ -690,25 +689,35 @@ def save(self):
         # or the interface wouldn't appear once IP was changed.
         retval = super(GlobalConfigurationForm, self).save()
 
-        whattoreload = "hostname"
+        whattoreload = ["hostname"]
         if (
             self.instance._orig_gc_domain != self.cleaned_data.get('gc_domain') or
             self.instance._orig_gc_nameserver1 != self.cleaned_data.get('gc_nameserver1') or
             self.instance._orig_gc_nameserver2 != self.cleaned_data.get('gc_nameserver2') or
             self.instance._orig_gc_nameserver3 != self.cleaned_data.get('gc_nameserver3')
         ):
             # Note notifier's _reload_resolvconf has reloading hostname folded in it
-            whattoreload = "resolvconf"
+            whattoreload.append("resolvconf")
         if (
             self.instance._orig_gc_ipv4gateway != self.cleaned_data.get('gc_ipv4gateway') or
             self.instance._orig_gc_ipv6gateway != self.cleaned_data.get('gc_ipv6gateway')
         ):
             # this supersedes all since it has hostname and resolvconf reloads folded in it
-            whattoreload = "networkgeneral"
+            whattoreload.append("networkgeneral")
             with client as c:
                 c.call('routes.sync')
 
-        notifier().reload(whattoreload)
+        if hasattr(notifier, 'failover_licensed') and notifier().failover_licensed() and \
+                self.instance._orig_gc_hostname_virtual != self.cleaned_data.get('gc_hostname_virtual'):
+            from freenasUI.services.models import services, NFS
+            svcobj = services.objects.get(srv_service='nfs')
+            nfsobj = NFS.objects.all()[0]
```

Comment:
I dont think we need to check if the NFS service is running.

service mountd reload will fail if the service is not enabled (because missing rc.conf vars)

Also there is no need to check `if nfsobj` since it is always True

Reply:
Why would you not check if NFS is running? If it isn't running, then it will be started when it wasn't meant to be ran. I'm not following. As for the extra check of nfsobj, having deal with numerous python tracebacks due to code like this, I just prefer to be extra careful.

Reply:
We do that everywhere. Just how it has been working for 7+ years. We blindly tell to restart SSH for instance, even if it is not enabled. NFS will not start if it is not enabled because of missing rc.conf.var as explained earlier.

As far as the check for nfsobj, you would hit IndexError way before (on .all()[0]) if nfs does not exist.

Reply:
will you still love me, tomorrow ?

Reply:
I am simply doing what I was asked for, you are free to ignore my advices :)

Reply:
Ignore you? never ;-) Not agree? sometimes ;-)